### PR TITLE
Fail the ingress test if it timesout getting address

### PR DESCRIPTION
**Pull request by nikhiljindal**
Updating the test to fail if it timesout getting IP address for the ingress rather than silently ignoring that error.
Also improved some logging to print more information.

This is to help in debugging tests added in https://github.com/kubernetes/kubernetes/pull/59234

cc @madhusudancs @MrHohn @nicksardo 

Ref https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/131

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/ingress_utils.go [hunk 1]
```diff
@@ -1172,7 +1172,7 @@ func (j *IngressTestJig) Update(update func(ing *extensions.Ingress)) {
 	for i := 0; i < 3; i++ {
 		j.Ingress, err = j.Client.ExtensionsV1beta1().Ingresses(ns).Get(name, metav1.GetOptions{})
 		if err != nil {
-			Failf("failed to get ingress %q: %v", name, err)
+			Failf("failed to get ingress %s/%s: %v", namespace, name, err)
```
Comment (MrHohn) [inline test/e2e/framework/ingress_utils.go]: here and below: namespace -> ns?

File: test/e2e/framework/ingress_utils.go [hunk 2]
```diff
@@ -1304,13 +1304,14 @@ func (j *IngressTestJig) WaitForIngressAddress(c clientset.Interface, ns, ingNam
...
 		if err != nil || len(ipOrNameList) == 0 {
-			j.Logger.Errorf("Waiting for Ingress %v to acquire IP, error %v", ingName, err)
+			j.Logger.Errorf("Waiting for Ingress %s/%s to acquire IP, error: %v, ipOrNameList: %v", ns, ingName, err, ipOrNameList)
 			if testutils.IsRetryableAPIError(err) {
 				return false, nil
 			}
 			return false, err
 		}
 		address = ipOrNameList[0]
+		j.Logger.Infof("Found address: %s", address)
```
Comment (MrHohn) [inline test/e2e/framework/ingress_utils.go]: Could we keep the ingress ns/name in log: `Found address %s for ingress %s/%s`?

**Review by MrHohn — APPROVED** [review]
Looks good, thanks! Some minor comments.

File: test/e2e/framework/ingress_utils.go (hunk 1 above)
Comment (nikhiljindal) [reply test/e2e/framework/ingress_utils.go]: Done

File: test/e2e/framework/ingress_utils.go (hunk 2 above)
Comment (nikhiljindal) [reply test/e2e/framework/ingress_utils.go]: Done

**Review by nikhiljindal — COMMENTED** [review]
Thanks for the quick review. Fixed as per comments

**Comment by nikhiljindal** [comment]
Test only change.
Adding to 1.10

**Comment by nikhiljindal** [comment]
kind/bug
priority/critical-urgent
sig/sig-multicluster

**Review by madhusudancs — COMMENTED** [review]
/lgtm

**Comment by tpepper** [comment]
@nikhiljindal you mention "...to help in debugging tests added in #59234."  That PR doesn't indicate anything about problems being debugged.  Is there an open issue describing the situation?  While this is a test only change the release team's trying to minimize churn and focus resources on resolving a fairly short list of specific issues right now ahead of the 1.10 release.  Any additional context and justification for 1.10 inclusion you can share would be useful in weighing the situation.

**Comment by nikhiljindal** [comment]
There is additional context in https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/131.
I have added a new test recently which is failing. This PR adds a meaningful error message to help in debugging. This is a small test only change.

That said, it is not strictly required for 1.10 and I agree that getting a high quality 1.10 release is the highest priority. If the release team thinks it adds a risk, then feel free to remove it from the milestone.

**Pull request merged**